internal/proxy/hook_interceptor.go

```diff
@@ -17,38 +17,47 @@ import (
 
 var hoo hook.Hook
 
+type someHandler func(ctx context.Context, req any, userName, fullMethod string, handler grpc.UnaryHandler) (interface{}, error)
+
 func UnaryServerHookInterceptor() grpc.UnaryServerInterceptor {
-	hookutil.InitOnceHook()
-	hoo = hookutil.Hoo
 	return func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		return HookInterceptor()(ctx, req, getCurrentUser(ctx), info.FullMethod, handler)
```

Comment:
Perhaps it would be better to use the HookInterceptor return value as a method directly, because it can reduce one layer of function calls, like HookIntercept(xxx, xxx, xxx)